Sources/StreamChat/Database/DTOs/ChannelDTO.swift

```diff
@@ -19,6 +23,8 @@ class ChannelDTO: NSManagedObject {
     @NSManaged var defaultSortingAt: Date
     @NSManaged var updatedAt: Date
     @NSManaged var lastMessageAt: Date?
+    
+    @NSManaged var isBeingRead: Bool
```

Comment:
I wonder if we can avoid introducing a local flag but instead look for current user in `channel.watchers`?

cc @b-onc @tbarbugli

Reply:
I've just realised there's another solution. What if we keep track of all created `channelList/channel` controllers in `ChatClient`? 🤔 then, we don't need to set/reset any local flag on `ChannelDTO` but can get channels that are being read simply by iterating over `[Weak<ChannelController>].map(\.cid)`

Reply:
this way, the sync can stay in `ConnectionRecoveryUpdater`